app/src/main/java/com/example/taskyy/data/repositories/AuthRepositoryImpl.kt

```diff
@@ -25,9 +25,11 @@ class AuthRepositoryImpl @Inject constructor(
     override fun registerUser(userEntity: UserEntity) {
```

Comment:
I've commented on that in detail in the last PR, but here you're still coupling your domain layer to your data layer which isn't allowed. `AuthRepository` which is in domain knows about `UserEntity` which is in data. Only the other way of communication is allowed, that data would know something from domain. 
To fix that, you could create a class like this in domain, e.g.:
```
data class Registration(
    val fullName: String,
    val email: String,
    val password: String
)
```
and then pass that to this `registerUser` function instead of the entity. The entity doesn't feel like the right type here anyways as it contains the password which you would rather not want to save in a local database